### PR TITLE
HV-1196 Splitting up PropertyConstraintLocation into specific impls f…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…or fields and getter methods

https://hibernate.atlassian.net/browse/HV-1196

### CODE REVIEW

**Comment by gsmet** [comment]
@gunnarmorling there is a javadoc generation error on CI:
```
/var/lib/jenkins/workspace/hibernate-validator-PR/distribution/../engine/src/main/java/org/hibernate/validator/internal/metadata/location/ConstraintLocation.java:94: error: reference not found
	 * e.g. a bean would be passed for a {@link PropertyConstraintLocation} but an object array for a
```

**Comment by gunnarmorling** [comment]
Thanks, pushed an update.

File: engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java [hunk 1]
```diff
@@ -1415,14 +1416,15 @@ private Object getCascadableValue(ValidationContext<?> validationContext, Object
...
-			ConstraintLocation delegate = ( (TypeArgumentConstraintLocation) location ).getDelegate();
-			if ( delegate instanceof PropertyConstraintLocation ) {
-				return ( (PropertyConstraintLocation) delegate ).getPropertyName();
-			}
+
+		if ( location instanceof FieldConstraintLocation ) {
+			return ( (FieldConstraintLocation) location ).getPropertyName();
+		}
+		else if ( location instanceof GetterConstraintLocation ) {
+			return ( (GetterConstraintLocation) location ).getPropertyName();
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java]: Nice cleanup.

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request closed without merging**